src/main/java/bio/terra/stairway/RetryRuleRandomBackoff.java

```diff
@@ -0,0 +1,48 @@
+package bio.terra.stairway;
+
+import java.util.concurrent.ThreadLocalRandom;
+import java.util.concurrent.TimeUnit;
+
+public class RetryRuleRandomBackoff implements RetryRule {
+    private long operationIncrementMilliseconds;
```

Comment:
It looks like all fields except `retryCount` can be `final` Making fields final makes it easier for a reader to know what state is expected to change over time, which IMO makes reading and reasoning about code easier as well.

Reply:
I made these final. Final has seemed like useless clutter to me. A throwback to a time when the compiler wasn't smart enough to know what was constant.

But I guess if it helps someone understand the code, that is a good enough reason to do it.